pkg/components/beyla.go

```diff
@@ -70,15 +73,24 @@ func RunBeyla(ctx context.Context, cfg *beyla.Config) error {
 func setupAppO11y(ctx context.Context, ctxInfo *global.ContextInfo, config *beyla.Config) error {
 	slog.Info("starting Beyla in Application Observability mode")
 
-	wg := sync.WaitGroup{}
-	defer wg.Wait()
-
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
 	instr := appolly.New(ctx, ctxInfo, config)
-	if err := instr.FindAndInstrument(&wg); err != nil {
+	if finderDone, err := instr.FindAndInstrument(); err != nil {
 		slog.Debug("can't find  target process", "error", err)
 		return fmt.Errorf("can't find target process: %w", err)
+	} else {
+		defer func() {
+			select {
+			case <-finderDone:
+				// ok!! setupAppO11y function ends
+			case <-time.After(beylaStopTimeout):
+				slog.Warn("timeout waiting for FindAndInstrument to finish. Some eBPF probes might remain loaded")
+			}
+		}()
```

Comment:
Not sure if we should maybe just do`<-finderDone` and let Kubernetes or service managers to terminate Beyla, instead.

Reply:
Currently, nothing hangs the shutdown process, it was more a preventive measure. But you are right that if it happens, we won't notice it. I'll remove the timeout.